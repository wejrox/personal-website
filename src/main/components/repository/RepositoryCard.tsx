import * as React from "react";
import Row from "reactstrap/lib/Row";
import Col from "reactstrap/lib/Col";

/**
 * @property {string} title        Title of the repository project.
 * @property {string} url          Link to the repository.
 * @property {string} description  Long description of the project within the repository.
 * @property {string} image        An image representing the project contained in the repository.
 */
interface IRepositoryCardProps {
    title: string;
    url: string;
    description: string;
    image: string;
}

/**
 * A box containing details about a git repository.
 */
class RepositoryCard extends React.Component<IRepositoryCardProps> {

    public render() {
        return (
            <Row>
                <Col>
                    Repository card
                </Col>
            </Row>
        );
    }
}

export default RepositoryCard;
