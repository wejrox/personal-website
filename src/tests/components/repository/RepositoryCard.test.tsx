import * as Enzyme from "enzyme";
import * as React from "react";
import RepositoryCard from "../../../main/components/repository/RepositoryCard";

describe("RepositoryCard Component", () => {
    const wrapper = Enzyme.shallow(<RepositoryCard description={""} image={""} title={""} url={""}/>);

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
