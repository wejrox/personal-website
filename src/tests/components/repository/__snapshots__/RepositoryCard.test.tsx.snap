// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard Component renders correctly 1`] = `ShallowWrapper {}`;
